exercises/minesweeper/src/example/java/Minesweeper.java

```diff
@@ -0,0 +1,89 @@
+import java.util.*;
+import java.util.stream.Collectors;
+
+public final class Minesweeper {
+
+    private static final char MINE_CHAR = '*';
+
+    private static final char SPACE_CHAR = ' ';
+
+    public static List<String> computeAnnotatedBoardFor(final List<String> inputBoard) throws IllegalArgumentException {
+        validateInputBoard(inputBoard);
+
+        final int numberOfRows = inputBoard.size();
+        final int numberOfColumns = inputBoard.isEmpty() ? 0 : inputBoard.get(0).length();
+
+        final List<String> result = new ArrayList<>();
+
+        for (int row = 0; row < numberOfRows; row++) {
+            String annotatedBoardRow = "";
+
+            for (int column = 0; column < numberOfColumns; column++) {
+                // If (row, col) is a mine, we're done.
+                if (inputBoard.get(row).charAt(column) == MINE_CHAR) {
+                    annotatedBoardRow += MINE_CHAR;
+                    continue;
+                }
+
+                // Compute row and column ranges to inspect (respecting board edges).
+                final int minRowToInspect = Math.max(row - 1, 0);
+                final int maxRowToInspect = Math.min(row + 1, numberOfRows - 1);
+                final int minColToInspect = Math.max(column - 1, 0);
+                final int maxColToInspect = Math.min(column + 1, numberOfColumns - 1);
+
+                int mineCount = 0;
+
+                // Count mines in the squares surrounding (row, col).
+                for (int rowToInspect = minRowToInspect; rowToInspect <= maxRowToInspect; rowToInspect++) {
+                    for (int colToInspect = minColToInspect; colToInspect <= maxColToInspect; colToInspect++) {
+                        if (inputBoard.get(rowToInspect).charAt(colToInspect) == MINE_CHAR) {
+                            mineCount += 1;
+                        }
+                    }
+                }
+
```

Comment:
@stkent this is really the only part of this method that makes me wish for refactoring.  I find myself wondering if there's not some way to grab the entire subset of characters concerned and reduce them to a number without nested `for` loops and `if`s.

Otherwise this PR is generally very nice work.